### PR TITLE
Fix validation and add convergence test

### LINKED ISSUE #1875: Incorrect model evaluation results in xgboost
During xgboost  model training, when setting the validation set, the evaluation result of the validation set is the same as the training set.

The reproducing procedure is shown as follows:

```sql
%%sqlflow
SELECT * FROM boston.train
TO TRAIN xgboost.gbtree
WITH
    objective="reg:squarederror",
    train.num_boost_round = 20,
    validation.select = "select * from boston.test;"
LABEL medv
INTO sqlflow_models.my_xgb_regression_model;

```
The output message:
```text
[0]	train-rmse:17.1167	validate-rmse:17.1167

[1]	train-rmse:12.3003	validate-rmse:12.3003

[2]	train-rmse:8.95505	validate-rmse:8.95505

[3]	train-rmse:6.57081	validate-rmse:6.57081

[4]	train-rmse:4.8891	validate-rmse:4.8891

[5]	train-rmse:3.69686	validate-rmse:3.69686

[6]	train-rmse:2.88164	validate-rmse:2.88164

[7]	train-rmse:2.30785	validate-rmse:2.30785

[8]	train-rmse:1.92918	validate-rmse:1.92918

[9]	train-rmse:1.62107	validate-rmse:1.62107

[10]	train-rmse:1.42948	validate-rmse:1.42948

[11]	train-rmse:1.30048	validate-rmse:1.30048

[12]	train-rmse:1.19441	validate-rmse:1.19441

[13]	train-rmse:1.09655	validate-rmse:1.09655

[14]	train-rmse:1.03565	validate-rmse:1.03565

[15]	train-rmse:0.981582	validate-rmse:0.981582

[16]	train-rmse:0.932085	validate-rmse:0.932085

[17]	train-rmse:0.867206	validate-rmse:0.867206

[18]	train-rmse:0.840945	validate-rmse:0.840945

[19]	train-rmse:0.799894	validate-rmse:0.799894
```

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #1875

### CODE REVIEW

File: python/sqlflow_submitter/xgboost/train.py [hunk 1]
```diff
@@ -72,13 +72,14 @@ def train(datasource,
...
     if len(validation_select.strip()) > 0:
-        dvalidate = xgb_dataset(datasource, 'validate.txt', select,
+        dvalidate = xgb_dataset(datasource, 'validate.txt', validation_select,
                                 feature_metas, feature_column_names,
                                 label_meta, is_pai, pai_validate_table)
         watchlist.append((dvalidate, "validate"))
 
     re = dict()
     print("Start training XGBoost model...")
+    print("model params: ", model_params)
```
Comment (typhoonzero) [inline python/sqlflow_submitter/xgboost/train.py]: Can remove this debug print here.

**Review by typhoonzero — DISMISSED** [review]

**Review by typhoonzero — APPROVED** [review]
